proj-android/PowerAuthLibrary/src/main/java/io/getlime/security/powerauth/networking/interceptors/HttpRequestInterceptor.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2018 Wultra s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getlime.security.powerauth.networking.interceptors;
+
+import android.support.annotation.NonNull;
+
+import java.net.HttpURLConnection;
+
+/**
+ * The {@code HttpRequestInterceptor} defines interface for modifying HTTP requests
+ * before their execution.
+ */
+public interface HttpRequestInterceptor {
+
+    /**
+     * Called before the provided {@link HttpURLConnection} is executed. The implementation must
+     * count with that method is called from other than UI thread.
+     *
+     * @param connection {@link HttpURLConnection} object to be modified.
+     */
+    void processRequestConnection(@NonNull HttpURLConnection connection);
```

Comment:
So do we really intend to allow any kind manipulation with `HttpUrlConnection` (thus manipulation with `HttpsUrlConnection` too)?
Or do we want to allow only header addition?

This way everything on `Http(s)UrlConnection` can be touched and changed such as `SSLSocketFactory`, `HostnameVerifier`, request properties (timeouts, method, …), …
We are entering dangerous waters with this, devs can easily break things. Is it ok for us?

Reply:
Well, my original idea was to add just a dictionary with an additional headers to the configuration, but @petrdvorak wanted this more "enterprise" :) It is the same on iOS, so yes, if our clients wants to shoot themselves in the foot, then lets allow that.

Reply:
To be clear. This approach allows us to add a more advanced tweaks in the future, but we should cover this in the documentation, that creating custom interceptors and tweaking the connection is not recommended.

Reply:
I've added warnings to inline documentation.